### ISSUE TITLE
[feature request ] integrate sync between users that want to run local ONLY.

### ISSUE THREAD

**Issue by cocoonkid** [gristlabs/grist-core#139]
I successfully sync sqlite db's from other apps via syncthing but I am certain this will not work with grist since its use cases have a lot more writes than reads.  I will try tho and report back.

Maybe this can be achieved via https://github.com/benbjohnson/litestream ?
 
Such a feature would be so helpful. But I understand this use-case might be too specific and a local server is the way to go.

**Comment by paulfitz** [comment]
litestream is a good candidate for backing up or mirroring Grist documents, but you'd hit trouble if the document is being edited by multiple users independently.

Likewise for syncthing. You'd need a merge method (and there isn't one yet https://github.com/gristlabs/grist-core/issues/40), and something to run it on conflicts. It looks like you can arrange for syncthing to preserve the common ancestor of a conflict https://www.rafa.ee/articles/resolve-syncthing-conflicts-using-three-way-merge/ but I don't see a way to actually set up conflict resolution. Also, if a Grist document changes on disk, you'd currently need to restart a Grist instance that is accessing it (the instance assumes it has complete control). That is something that could be tweaked though.

In summary - yes a local server is the way to go :) - agree syncing would be neat though!

**Comment by cocoonkid** [comment]
@paulfitz 

Thank you for taking the time to reply in such a thorough  way.

Highly appreciated. ❤️
